onnxruntime/core/providers/webgpu/buffer_manager.cc

```diff
@@ -297,11 +309,31 @@ WGPUBuffer BufferManager::Create(size_t size, wgpu::BufferUsage usage) {
   return buffer;
 }
 
+WGPUBuffer BufferManager::CreateUMA(size_t size, wgpu::BufferUsage usage) {
+  ORT_ENFORCE(usage & wgpu::BufferUsage::Storage, "UMA buffer must have storage usage.");
+  auto& cache = GetCacheManager(static_cast<WGPUBufferUsage>(usage));
+  auto buffer_size = cache.CalculateBufferSize(size);
+
+  wgpu::BufferDescriptor desc{};
+  desc.size = buffer_size;
+  // Ensure the buffer is mapped for writing at creation.
+  desc.usage = usage | wgpu::BufferUsage::MapWrite;
+  desc.mappedAtCreation = true;
+  auto buffer = context_.Device().CreateBuffer(&desc).MoveToCHandle();
+
+  ORT_ENFORCE(buffer, "Failed to create GPU buffer: size=", buffer_size, ", usage=", uint64_t(usage), ".");
+
+  cache.RegisterBuffer(buffer, size);
+  return buffer;
+}
+
 void BufferManager::Release(WGPUBuffer buffer) {
+  ORT_ENFORCE(wgpuBufferGetMapState(buffer) == WGPUBufferMapState_Unmapped, "Buffer must be unmapped before releasing.");
```

Comment:
This is trivial in native but going to be expensive in wasm. Do you think it is a good idea to wrap it with validation level?

Reply:
Done!